components/script/layout_wrapper.rs

```diff
@@ -757,46 +779,48 @@ impl<'ln> ServoThreadSafeLayoutNode<'ln> {
     }
 }
 
+// NB: The implementation here is a bit tricky because elements implementing
+// pseudos are supposed to return false for is_element().
```

Comment:
Well, other browser elements put NAC in the DOM, which Servo explicitly avoids doing. But yeah, we might be able to make the intra-layout abstraction cleaner. Agreed on out of scope.